src/core/ddsc/tests/psmxif.c

```diff
@@ -0,0 +1,441 @@
+// Copyright(c) 2020 to 2023 ZettaScale Technology and others
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+// v. 1.0 which is available at
+// http://www.eclipse.org/org/documents/edl-v10.php.
+//
+// SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+
+#include <assert.h>
+#include <limits.h>
+
+#include "dds/ddsrt/mh3.h"
+#include "dds/ddsrt/md5.h"
+#include "dds/ddsrt/io.h"
+#include "dds/ddsrt/heap.h"
+#include "dds/ddsrt/string.h"
+#include "dds/ddsrt/bswap.h"
+#include "dds/ddsrt/environ.h"
+#include "dds/ddsrt/static_assert.h"
+
+#include "dds/dds.h"
+#include "dds/ddsi/ddsi_entity_index.h"
+#include "ddsi__addrset.h"
+#include "ddsi__entity.h"
+#include "ddsi__xevent.h"
+#include "dds__entity.h"
+#include "dds__serdata_default.h"
+
+#include "config_env.h"
+#include "test_common.h"
+#include "psmx_dummy_public.h"
+#include "Array100.h"
+#include "DynamicData.h"
+#include "PsmxDataModels.h"
+
+static void free_strings(uint32_t len, char** strings)
+{
+    if (len != 0 && strings != NULL) {
+        for (uint32_t i = 0; i < len; i++) {
+            dds_free(strings[i]);
+        }
+    }
+    dds_free(strings);
+}
+
+/**
+ * @brief Create a participant object, providing a configuration to choose the psmx interface.
+ * 
+ * @param[in] int_dom the domain id
+ * @param[in] cdds_psmx_name the name of the psmx interface to use
+ * @param[in] locator an array of 16 bytes, NULL if not used
+ * @return dds_entity_t the participant
+ */
+static char* create_config(dds_domainid_t int_dom, const char* cdds_psmx_name, const uint8_t* locator)
+{
+  char *configstr;
+  char locator_str[74];
+  if ( locator == NULL ) {
```

Comment:
These complications with locators are no longer used in any meaningful way, they should be removed. A single, fixed, configuration string suffices for the purpose of the test.